File: 11.container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max_area = 0;
        int i = 0, j = height.size() - 1;
        while (i < j) {
            max_area = max(max_area, min(height[i], height[j]) * (j - i));
            height[i] > height[j] ? j-- : i++;
        }
        return max_area;
    }
};

File: 11.container-with-most-water.py
class Solution:
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        max_area = 0;
        i, j = 0, len(height) - 1
        while i < j:
            max_area = max(max_area, min(height[i], height[j]) * (j - i));
            if height[i] > height[j]:
                j -= 1
            else:
                i += 1
        return max_area;